src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -230,7 +260,90 @@ def excepthook(args):
+    Runs a simulated test multiple times with different given parameters
+
+    Result of the test is determined by comparing the number of
+    passing iterations to a predetermined acceptable threshold
+    """
+
+    def __int__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+
+    def run_test(
+        self,
+        setup=(lambda: None),
+        params=[],
+        ag_always_validation_sequence_set=[[]],
+        ag_eventually_validation_sequence_set=[[]],
+        test_timeout_s=3,
+        tick_duration_s=0.0166,  # Default to 60hz
+        **kwargs,
+    ):
+        """Run an aggregate test
+
+        :param setup: Function that sets up the World state and the gamecontroller before running the test
+        :param params: List of parameters for each iteration of the test
+        :param ag_always_validation_sequence_set: Validation functions for aggregate testing
+                                that should hold on every tick
+        :param ag_eventually_validation_sequence_set: Validation functions for aggregate testing
+                                that should eventually be true, before the test ends
+        :param test_timeout_s: The timeout for the test, if any eventually_validations
+                                remain after the timeout, the test fails.
+        :param tick_duration_s: The simulation step duration
+
+        """
+
+        threading.excepthook = self.excepthook
+
+        # Runs the test once for each given parameter
+        # Catches Assertion Error thrown by failing test and increments counter
+        # Calculates overall results and prints them
+        for x in range(len(params)):
+
+            setup(params[x])
+
+            failed_tests = 0
+
+            try:
+                if self.thunderscope:
+                    # If thunderscope is enabled, run the test in a thread and show
+                    # thunderscope on this thread. The excepthook is setup to catch
+                    # any test failures and propagate them to the main thread
+                    run_sim_thread = threading.Thread(target=self.runner, daemon=True)
+                    run_sim_thread.start()
+                    self.thunderscope.show()
+                    run_sim_thread.join()
+
+                    if self.last_exception:
+                        pytest.fail(str(self.last_exception))
+
+                # If thunderscope is disabled, just run the test
+                else:
+                    self.runner(
+                        ag_always_validation_sequence_set,
+                        ag_eventually_validation_sequence_set,
+                        test_timeout_s[x]
+                        if type(test_timeout_s) == list
+                        else test_timeout_s,
+                        tick_duration_s,
+                    )
+            except AssertionError:
+                failed_tests += 1
+
+        print(f"{failed_tests} test failed")
+
+        assert failed_tests == 0
```

Comment:
the pass/fail rate percentage will be addressed in another PR, correct?

Reply:
Yes once we work on the validation

Reply:
bump

Reply:
sorry made the issue but forgot to add a todo

Reply:
can this be a seperate PR for fixing the output? I added a TODO for that issue